RuntimeUnityEditor/Windows/ObjectTree/ObjectTreeViewer.cs

```diff
@@ -490,6 +503,8 @@ private void DisplayObjectTree()
 
                     if (notVisibleCount > 0)
                         GUILayout.Space(_singleObjectTreeItemHeight * notVisibleCount);
+                    if (_scrollTreeToSelected && Event.current.type == EventType.Repaint)
+                        _scrollTarget = Mathf.Min(_scrollTarget, _singleObjectTreeItemHeight * currentCount + _singleObjectTreeItemMargin - _objectTreeHeight);
```

Comment:
After all items have been displayed, this adjusts the _scrollTreeToSelected if the scrollTarget would be beyond the highest it can be, i.e. selecting an item when the scrollbar is already bottomed out.